model/CompatibilityChecker.php

```diff
@@ -14,101 +14,241 @@
+    public const COMPATIBILITY_COMPATIBLE = 1;
+    public const COMPATIBILITY_NOT_TESTED = 2;
+    public const COMPATIBILITY_SUPPORTED = 3;
+    public const COMPATIBILITY_NOT_SUPPORTED = 4;
+
     protected $compatibility;
+    protected $supported;
+    protected $excludedBrowsers;
+    protected $excludedOS;
 
     /**
-     * CompatibilityChecker constructor
-     * Check parameter required
      * Extract compatibility file
-     * @throws \tao_models_classes_FileNotFoundException
+     * @throws common_exception_FileSystemError
+     */
+    public function getCompatibilityList(): array
+    {
+        if ($this->compatibility == null) {
+            $compatibilityFile = __DIR__ . '/../include/compatibility.json';
+
+            if (!file_exists($compatibilityFile)) {
+                throw new common_exception_FileSystemError('Unable to find the compatibility file');
+            }
+            $this->compatibility = json_decode(file_get_contents($compatibilityFile), true);
+        }
+        return $this->compatibility;
+    }
+
+    /**
+     * Fetch the support list
+     * @throws common_exception_FileSystemError
+     * @throws common_exception_InconsistentData
+     * @throws common_exception_MissingParameter
+     */
+    public function getSupportedList(): array
+    {
+        if ($this->supported == null) {
+            $service = $this->getServiceLocator()->get(DiagnosticServiceInterface::SERVICE_ID);
+            $config = $service->getDiagnosticJsConfig();
+            $supportListUrl = $config['diagnostic']['testers']['browser']['browserslistUrl'];
+
+            if (!$supportListUrl) {
+                throw new common_exception_MissingParameter('The URL to the list of supported browser is not configured');
+            }
+
+            $supportedListData = @file_get_contents($supportListUrl);
+            if ($supportedListData === false) {
+                $error = error_get_last();
+                throw new \common_exception_FileSystemError(sprintf(
+                    'Unable to fetch the list of supported browsers due to error: %s',
+                    $error['message']
+                ));
+            }
+
+            $supportedList = json_decode($supportedListData, true);
+
+            if (!$supportedList) {
+                throw new common_exception_InconsistentData('Unable to decode list of supported browsers');
+            }
+
+            $this->supported = array_map(function ($entry) {
+                $entry['compatible'] = self::COMPATIBILITY_SUPPORTED;
+
+                $entry['versions'] = array_reduce($entry['versions'], function ($versions, $version) {
+                    return array_merge($versions, explode('-', $version));
+                }, []);
+
+                return $entry;
+            }, $supportedList);
+        }
+        return $this->supported;
+    }
+
+    protected function filterVersion($version): string
+    {
+        return preg_replace('#(\.0+)+($|-)#', '', $version);
```

Comment:
I think it was more or less intended, to workaround the weakness of the default comparison made by the native [version_compare](https://www.php.net/manual/en/function.version-compare.php) function. The native function states that `1 < 1.0.0`, instead of the expected `1 == 1.0.0`.
This workaround will states that `1.2-beta == 1.2beta`. While also ensuring than `1-beta == 1.0.0-beta`